agent/src/util/config_action.rs

```diff
@@ -276,6 +289,28 @@ async fn handle_config_delete(
     Ok(())
 }
 
+/// Checks to see if the configuration needs to be recreated.
+/// At present, this just checks to see if the `.metadata.generation` has changed.
+/// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+async fn should_recreate_config(
+    config: &Configuration,
+    config_map: ConfigMap,
+) -> Result<bool, anyhow::Error> {
+    let name = config.metadata.name.clone().unwrap();
+    let last_generation = config_map
+        .lock()
+        .await
+        .get(&name)
+        .ok_or_else(|| anyhow::anyhow!("Configuration {} not found in ConfigMap", name))?
```

Comment:
nit curiousity: if you use &name here, do you need the clone on line 299?

Reply:
I get this when I attempt to do that:

```
cannot move out of `config.metadata.name` which is behind a shared reference
move occurs because `config.metadata.name` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` traitrustcE0507
config_action.rs(299, 16): consider borrowing the `Option`'s content: `config.metadata.name.as_ref()`
```

```rust
let name = config.metadata.name.unwrap();
let last_generation = config_map
    .lock()
    .await
    .get(&name)
    .ok_or_else(|| anyhow::anyhow!("Configuration {} not found in ConfigMap", &name))?
    .last_generation;
```

Reply:
Looks like I could do this though...

```rust
let name = config.metadata.name.as_ref().unwrap();
let last_generation = config_map
    .lock()
    .await
    .get(name)
    .ok_or_else(|| anyhow::anyhow!("Configuration {} not found in ConfigMap", &name))?
    .last_generation;
```